js/services/cache/StateCaching.ts

```diff
@@ -44,29 +51,41 @@ export const pollForStateValues = (stateId: string, tenantId: string, token: str
 
     clearTimeout(timer);
 
-    if (!store.getState().isOffline) { // only poll api whilst online
-        const url = `${manywho.settings.global('platform.uri')}/api/run/1/state/${stateId}/values`;
-        const request = {
-            headers: {
-                ManyWhoTenant: tenantId,
-            },
-        };
-        if (authenticationToken) {
-            request.headers['Authorization'] = authenticationToken;
-        }
-        return fetch(url, request)
-            .then((response) => {
-                return response.json();
-            })
-            .then((response) => {
-                injectValuesIntoState(response);
-
-                // This is so the flow can perioically poll for the latest values
-                // in case values have changed from a user joining the state
-                timer = setTimeout(
-                    () => { pollForStateValues(stateId, tenantId, authenticationToken); }, pollInterval,
-                );
-            })
-            .catch(response => console.error(response));
+    const url = `${manywho.settings.global('platform.uri')}/api/run/1/state/${stateId}/values`;
```

Comment:
Are we always polling for state values, even when online? If so, could this potentially be a large payload? Should we poll a designated "health" type of endpoint and only fetch values when needed?

Reply:
The polling of state values whilst a flow has connectivity is a required mechanism for the offline experience to work as expected. Consider this use case - there is a flow with multiple forms to move through... user completes form A whilst online, then moves to form B. Whilst filling out form B, network is lost. Form B is completed whilst offline but the user now needs to move back to form A to make a change. The user expects form A's fields to be populated, but at this point the flow would know nothing about Form A's component values that had been set previously. The polling of state values whilst online ensures that the offline state container always gets updated with the latest values set in the flow. Polling also accounts for values being set by other users who happen to join the current state.
Since this is an existing required mechanism, I decided to use this as the network heartbeat check rather than having an additional polling mechanism.

Reply:
Ok, fair enough, it was just something that occurred to me. Also, I hadn't taken into account these are just all the values within a state, not all values in a tenant which is where we've seen problems before. Could be something to keep in mind though.